unit/goto-programs/restrict_function_pointers.cpp

```diff
@@ -148,3 +216,8 @@ TEST_CASE("Json conversion", "[core]")
     function_pointer_restrictions1.restrictions ==
     function_pointer_restrictions2.restrictions);
 }
+
+TEST_CASE("Get function pointer by name restrictions", "[core]")
+{
+  get_function_pointer_by_name_restrictions_test();
```

Comment:
:grey_question: Why not just put the function in the test? Doing so makes it easier to find the test, and the CLion integration of catch unit tests will work better

Reply:
The test accesses some protected methods of `function_pointer_restrictionst`. So to access those we declare `get_function_pointer_by_name_restrictions_test()` as a friend (see at the top of the file).